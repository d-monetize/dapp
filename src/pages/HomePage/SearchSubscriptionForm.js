import React, { useState } from "react"
import PropTypes from "prop-types"
import { Form, Input, Button, Icon } from "semantic-ui-react"
import styles from "./SearchSubscriptionForm.module.css"

export function SearchSubscriptionForm(props) {
  const [address, setAddress] = useState("")

  function onSubmit() {
    const addr = address.trim()

    if (addr) {
      props.onSubmit(addr)
    }
  }

  function onChangeAddress(e) {
    setAddress(e.target.value)
  }

  return (
    <Form onSubmit={onSubmit}>
      <div className={styles.row}>
        <div className={styles.input}>
          <Input
            placeholder="Find Subscription"
            onChange={onChangeAddress}
            value={address}
            autoFocus
            fluid
            size="huge"
          />
        </div>
        <Button icon size="huge">
          <Icon name="search" />
        </Button>
      </div>
    </Form>
  )
}

SearchSubscriptionForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
}

export default SearchSubscriptionForm
