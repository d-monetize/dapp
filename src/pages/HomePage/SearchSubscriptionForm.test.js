import React from "react"
import { shallow } from "enzyme"
import { SearchSubscriptionForm } from "./SearchSubscriptionForm"

const props = {
  onSubmit: jest.fn(),
}

beforeEach(() => {
  props.onSubmit.mockClear()
})

test("it renders", () => {
  const component = shallow(<SearchSubscriptionForm {...props} />)

  expect(component).toMatchSnapshot()
})

test("change address", () => {
  const component = shallow(<SearchSubscriptionForm {...props} />)

  component.find("Input").simulate("change", { target: { value: "foo" } })

  expect(component).toMatchSnapshot()
})

test("onSubmit", () => {
  const component = shallow(<SearchSubscriptionForm {...props} />)

  component.find("Input").simulate("change", { target: { value: "bar" } })
  component.find("Form").simulate("submit")

  expect(props.onSubmit.mock.calls.length).toEqual(1)
  expect(props.onSubmit.mock.calls[0][0]).toEqual("bar")
})

test("onSubmit should not be called if address is empty", () => {
  const component = shallow(<SearchSubscriptionForm {...props} />)

  component.find("Form").simulate("submit")

  expect(props.onSubmit.mock.calls.length).toEqual(0)
})
