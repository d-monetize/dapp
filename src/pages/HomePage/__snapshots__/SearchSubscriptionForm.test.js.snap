// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change address 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="input"
    >
      <Input
        autoFocus={true}
        fluid={true}
        onChange={[Function]}
        placeholder="Find Subscription"
        size="huge"
        type="text"
        value="foo"
      />
    </div>
    <Button
      as="button"
      icon={true}
      size="huge"
    >
      <Icon
        as="i"
        name="search"
      />
    </Button>
  </div>
</Form>
`;

exports[`it renders 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="input"
    >
      <Input
        autoFocus={true}
        fluid={true}
        onChange={[Function]}
        placeholder="Find Subscription"
        size="huge"
        type="text"
        value=""
      />
    </div>
    <Button
      as="button"
      icon={true}
      size="huge"
    >
      <Icon
        as="i"
        name="search"
      />
    </Button>
  </div>
</Form>
`;
